webpack/components/Table/TableHooks.js

```diff
@@ -41,14 +42,143 @@ class ReactConnectedSet extends Set {
       this.delete(id);
     }
   }
+
+  addAll(ids) {
+    ids.forEach(id => super.add(id));
+    this.forceRender();
+  }
 }
 
-const useSet = (initialArry) => {
+export const useSet = (initialArry) => {
   const [, setToggle] = useState(0);
   // needed because mutating a Ref won't cause React to rerender
   const forceRender = () => setToggle(prev => prev + 1);
   const set = useRef(new ReactConnectedSet(initialArry, forceRender));
   return set.current;
 };
 
-export default useSet;
+export const useSelectionSet = (results, metadata, initialArry = [], idColumn = 'id') => {
+  const selectionSet = useSet(initialArry);
+  const ids = results?.map(result => result[idColumn]) ?? [];
+  const areAllRowsOnPageSelected = () => Number(ids?.length) > 0 &&
+                                         ids.every(result => selectionSet.has(result));
+
+  const selectPage = () => selectionSet.addAll(ids);
+  const selectNone = () => selectionSet.clear();
+  const onRowSelect = (isSelected, id) => {
+    if (isSelected) {
+      selectionSet.add(id);
+    } else {
+      selectionSet.delete(id);
+    }
+  };
+
+  const selectedCount = selectionSet.size;
+
+  const isSelected = id => selectionSet.has(id);
+
+  return {
+    onRowSelect,
+    selectedCount,
+    areAllRowsOnPageSelected,
+    selectPage,
+    selectNone,
+    isSelected,
+    selectionSet,
+  };
+};
+
+export const useBulkSelect = (results, metadata, initialArry = [], idColumn = 'id') => {
```

Comment:
`useBulkSelect` and `useSelectionSet` are so important that I think they warrant their own Jest tests.  Or, make sure to thoroughly test all functionality via the ErrataTab tests.

Reply:
Agreed. Will work on em with your help tomorrow :)